compiler/resolution/functionResolution.cpp

```diff
@@ -5184,81 +5194,101 @@ resolveNew(CallExpr* call)
         if (ct->initializerStyle == DEFINES_INITIALIZER) {
           toReplace->replace(new UnresolvedSymExpr("init"));
           initCall = true;
+
         } else {
-          toReplace->replace(new UnresolvedSymExpr(ct->defaultInitializer->name));
+          FnSymbol* ctInit = ct->defaultInitializer;
+
+          toReplace->replace(new UnresolvedSymExpr(ctInit->name));
         }
       }
     }
 
     // 2: move the 1st argument into the baseExpr
     // 3: make it a normal call and not a PRIM_NEW
     Expr* arg = call->get(1);
+
     arg->remove();
+
     call->primitive = NULL;
-    call->baseExpr = arg;
+    call->baseExpr  = arg;
+
     parent_insert_help(call, call->baseExpr);
 
     if (initCall) {
       // 4: insert the allocation for the instance and pass that in as an
       // argument if we're making a call to an initializer
-      Type* typeToNew = toReplace->symbol()->typeInfo();
-      VarSymbol* new_temp = newTemp("new_temp", typeToNew);
+      Type*      typeToNew = toReplace->symbol()->typeInfo();
+      VarSymbol* newTmp    = newTemp("new_temp", typeToNew);
+
+      VarSymbol* newMT     = newTemp("_mt",      dtMethodToken);
+
       if (typeToNew->symbol->hasFlag(FLAG_GENERIC)) {
-        USR_FATAL(call, "Sorry, new style initializers don't work with generics yet.  Stay tuned!");
+        USR_FATAL(call,
+                  "Sorry, new style initializers don't work with "
+                  "generics yet.  Stay tuned!");
       }
-      DefExpr* def = new DefExpr(new_temp);
-      Expr* insertBeforeMe = ((!isBlockStmt(call->parentExpr)) ?
-                              call->parentExpr : call);
+
+      DefExpr* def            = new DefExpr(newTmp);
+      Expr*    insertBeforeMe = NULL;
+
+      if (isBlockStmt(call->parentExpr) == true) {
+        insertBeforeMe = call;
+
+      } else {
+        insertBeforeMe = call->parentExpr;
+      }
+
       insertBeforeMe->insertBefore(def);
+
```

Comment:
I would prefer you didn't do this change.  While it makes the code more pleasing to the eye, it will cause merge conflicts with my branch without improving functionality.  I would be fine with this change occurring after my merge.